### PR TITLE
set http errorcode if an error occured; differend redis cachetimes for errorpages

### LINKED ISSUE #321: Returns http status code 200 on error
Currently there is a problem with the Official-RKI-API.
The Official-RKI-API returns error code 400, but this api returns status code 200, when it shouldn't, because there is obviously a problem.

grafik

### PR DESCRIPTION
**Pull request by Rubber1Duck**
Changed the http-status for RKIErrors to 503 and axios errors to 503 too. All other errors are set to 400.
Changed the redis caching time for errormessages too. Only succsessfull requests (http status 200) are cached for 1800 sec now.
Fix #321 
I removed the parameter checks!

### CODE REVIEW

**Comment by Rubber1Duck** [comment]
@marlon360 i have removed the parameter checks! Only the passing through of the RKI error to the http error remained, as well as different cache times for different http statuses. Successful requests (http status 200) are cached for 1800 seconds as before. All other http statuses are only cached for 180 seconds! please review!

File: src/responses/germany.ts [hunk 1]
```diff
@@ -168,7 +173,12 @@ export async function GermanyCasesHistoryResponse(
 export async function GermanyWeekIncidenceHistoryResponse(
   days?: number
 ): Promise<GermanyHistoryData<{ weekIncidence: number; date: Date }>> {
-  if (days != null) {
```
Comment (marlon360) [inline src/responses/germany.ts:171]: @Rubber1Duck Is it save to remove the null check? Because `0 === false`.

Comment (Rubber1Duck) [reply src/responses/germany.ts:171]: the null check is still there! see line 181

Comment (Rubber1Duck) [reply src/responses/germany.ts:171]: @marlon360 oh mann, sorry i now understand what you meen! will check this, but i think you are right!

Comment (Rubber1Duck) [reply src/responses/germany.ts:171]: A request with parameter days = 0 without a check for null, as here done will response the full set of data from 2020-01-02 as you aspected! with a check for null (i.e. days !=0) is the response an empty array! The Null check alone is not the right solution. The "0" as a possible parameter for days should therefore be excluded! will patch this soon!

Comment (Rubber1Duck) [reply src/responses/germany.ts:171]: done

**Review by marlon360 — APPROVED** [review]

**Pull request merged**